lib/rubocop/cop/rspec/subject_stub.rb

```diff
@@ -74,71 +75,59 @@ class SubjectStub < Cop
         PATTERN
 
         def on_block(node)
+          # process only describe/context/... example groups
           return unless example_group?(node)
 
-          find_subject_stub(node) do |stub|
+          # skip already processed example group
+          # it's processed if is nested in one of the processed example groups
+          return unless (processed_example_groups & node.ancestors).empty?
+
+          # add example group to already processed
+          processed_example_groups << node
+
+          # find all custom subjects e.g. subject(:foo) { ... }
+          named_subjects = find_all_named_subjects(node)
+
+          # look for method expectation matcher
+          find_subject_expectations(node, named_subjects) do |stub|
             add_offense(stub)
           end
         end
 
         private
 
-        # Find subjects within tree and then find (send) nodes for that subject
-        #
-        # @param node [RuboCop::Node] example group
-        #
-        # @yield [RuboCop::Node] message expectations for subject
-        def find_subject_stub(node, &block)
-          find_subject(node) do |subject_name, context|
-            find_subject_expectation(context, subject_name, &block)
-          end
+        def processed_example_groups
+          @processed_example_groups ||= Set[]
```

Comment:
Is there a difference with `Set.new`?

Reply:
I believe there is no any difference.This form is just shorter.

Reply:
Maybe use the more commonly used (at least from my standpoint, I've never seen `Set[]` so far) to avoid confusion?